core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -909,6 +774,237 @@ private void initialize() {
+            }
+
+            // Change the state of complete batch since the same state exists for the entire inFlight batch.
+            // The member id is reset to EMPTY_MEMBER_ID irrespective of the acknowledge type as the batch is
+            // either released or moved to a state where member id existence is not important. The member id
+            // is only important when the batch is acquired.
+            InFlightState updateResult = inFlightBatch.startBatchStateTransition(
+                recordState,
+                false,
+                this.maxDeliveryCount,
+                EMPTY_MEMBER_ID
+            );
+            if (updateResult == null) {
+                log.debug("Unable to acknowledge records for the batch: {} with state: {}"
+                        + " for the share partition: {}-{}", inFlightBatch, recordState, groupId,
+                    topicIdPartition);
+                return Optional.of(
+                    new InvalidRecordStateException("Unable to acknowledge records for the batch"));
+            }
+
+            // Successfully updated the state of the batch.
+            updatedStates.add(updateResult);
+            stateBatches.add(
+                new PersisterStateBatch(inFlightBatch.firstOffset, inFlightBatch.lastOffset,
+                    updateResult.state.id, (short) updateResult.deliveryCount));
+
+            // If the maxDeliveryCount limit has been exceeded, the record will be transitioned to ARCHIVED state.
+            // This should not change the nextFetchOffset because the record is not available for acquisition
+            if (recordState == RecordState.AVAILABLE
+                && updateResult.state != RecordState.ARCHIVED) {
+                findNextFetchOffset.set(true);
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+        return Optional.empty();
+    }
+
+    private void rollbackOrProcessStateUpdates(
+        Throwable throwable,
+        List<InFlightState> updatedStates,
+        List<PersisterStateBatch> stateBatches
+    ) {
+        lock.writeLock().lock();
+        try {
+            if (throwable != null || !isWriteShareGroupStateSuccessful(stateBatches)) {
+                // Log in DEBUG to avoid flooding of logs for a faulty client.
+                log.debug("Request failed for updating state, rollback any changed state"
+                    + " for the share partition: {}-{}", groupId, topicIdPartition);
+                updatedStates.forEach(state -> state.completeStateTransition(false));
+            } else {
+                log.trace("State change request successful for share partition: {}-{}",
+                    groupId, topicIdPartition);
+                updatedStates.forEach(state -> {
+                    state.completeStateTransition(true);
+                    // Cancel the acquisition lock timeout task for the state since it is released successfully.
```

Comment:
comment can be changed to "Cancel the acquisition lock timeout task for the state since it is acknowledged or released successfully on session close."

Reply:
Done.